clientconn.go

```diff
@@ -137,7 +138,9 @@ func (dcs *defaultConfigSelector) SelectConfig(rpcInfo iresolver.RPCInfo) (*ires
 func DialContext(ctx context.Context, target string, opts ...DialOption) (conn *ClientConn, err error) {
 	cc := &ClientConn{
 		target: target,
-		csMgr:  &connectivityStateManager{},
+		csMgr: &connectivityStateManager{
+			pubSub: grpcsync.NewPubSub(),
+		},
```

Comment:
Sorry, I probably should have been more explicit. What I meant was to have a function to create the `connectivityStateManager`.

```golang
func newConnectivityStateManager(id *channelz.Identifier) *connectivityStateManager {
  return &connectivityStateManager{
    channelzID: id,
    pubSub: grpcsync.NewPubSub(),
  },
}
```

And in this function, instead of initializing the `csMgr` field here, wait until the call to `cc.channelzRegistration(target)` down below.

Also, change `cc.channelzRegistration()` to not set `cc.csMgr.channelzID = cc.channelzID`.